File: JavaScript/repeticoes.js
function repeticoes1() {
  let listaJogadores = [];
  for (index = 6; listaJogadores.length < index; ) {
    let jogador = {
      nome: prompt("Insira o nome do jogador."),
      idade: Number(prompt("Insira a idade.")),
    };
    listaJogadores.push(jogador);
  }

  listaOrdem = listaJogadores.sort((a, b) => a.idade - b.idade);
  listaOrdem.forEach((jogador) => {
    console.log(
      "Jogador com nome " +
        jogador.nome +
        " tem a idade de " +
        jogador.idade +
        " Anos"
    );
  });
}

// repeticoes1();

function repeticoes2() {
  let numeroDigitado = 1;
  while (numeroDigitado != 0) {
    console.log("Número digitado. " + numeroDigitado);
    numeroDigitado = Number(prompt("Digite um número."));
  }
}

// repeticoes2();

function repeticoes3() {
  let resposta = "s";
  while (resposta === "s") {
    let valorCompra = Number(prompt("Digite o valor da compra."));
    let valorDinheiro = Number(
      prompt("Digite o valor em dinheiro que você vai passar.")
    );
    do {
      let troco = valorDinheiro - valorCompra;
      console.log(
        "A compra do cliente deu R$" +
          valorCompra +
          " e valor passado foi R$" +
          valorDinheiro +
          ". O troco foi de R$" +
          troco
      );
    } while (valorCompra >= valorDinheiro);

    resposta = prompt(
      "Você ira atender outro cliente? 's' para Sim e 'n' para Não "
    );
  }
}

repeticoes3();

File: Python/repeticoes.py
def repeticoes1():
  jogadores = []
  for i in range(6):
    jogador = {
      'nome': input("Digite o nome do Jogador: "),
      'idade': int(input("Digite o idade do Jogador: "))
    }
    jogadores.append(jogador)

  def idadeOrdenada(e):
    return e['idade']

  jogadores.sort(key=idadeOrdenada)
  for jogador in jogadores:
    print("Jogador com nome", jogador['nome'], "tem a idade de", jogador['idade'], "anos")


# repeticoes1()

def repeticoes2():
  alunos = []
  media_alunos = 0
  qtd_alunos = int(input("Quantos alunos tem na sala: "))
  for i in range(qtd_alunos):
    aluno = {
      "nome": input("Informe o nome do aluno "),
      "nota": float(input("Informe a nota do aluno "))
    }
    media_alunos = media_alunos + aluno['nota']
  media_alunos = media_alunos / qtd_alunos
  print("A média dos alunos foi de", media_alunos)

# repeticoes2()

def repeticoes3():
  numero_digitado = float(input("Digite um número: "))
  while numero_digitado >= 0:
    numero_digitado = float(input("Digite um número: "))

# repeticoes3()

def repeticoes4():
  resposta = 's'
  while resposta == 's':
    valor_compra = float(input("Digite o valor da sua compra: "))
    valor_passado = float(input("Digite o valor em dinheiro para passar: "))
    if valor_compra <= valor_passado:
      troco = valor_passado - valor_compra
      print("O seu troco será de", "R$", troco)
    else:
      print("Você não possui o dinheiro suficienta para a compra!")
      break
    resposta = input("Outra compra? Digite 's' para Sim e 'n' para Não")

# repeticoes4()